model_analyzer/result/result_manager.py

```diff
@@ -465,430 +212,141 @@ def top_n_results(self, model_name=None, n=-1, include_default=False):
-        ]
-        max_batch_sizes = [
-            model_config.max_batch_size() for model_config in model_configs
-        ]
-        dynamic_batchings = [
-            model_config.dynamic_batching_string()
-            for model_config in model_configs
-        ]
-        cpu_onlys = [
-            run_config_result.run_config().cpu_only()
-            for model_config in model_configs
-        ]
-        backend_parameters = [
-            model_config._model_config.parameters
-            for model_config in model_configs
-        ]
+        _update_stats(result_stats, self._across_model_sorted_results,
+                      TOP_MODELS_REPORT_KEY)
 
-        return instance_groups, max_batch_sizes, dynamic_batchings, cpu_onlys, backend_parameters
+        return result_stats
 
-    def _tabulate_measurement(self, model_name, instance_groups,
-                              max_batch_sizes, dynamic_batchings,
-                              run_config_measurement, passes, cpu_onlys,
-                              backend_parameters):
+    def _init_state(self):
         """
-        Add a single RunConfigMeasurement to the specified
-        table
+        Sets ResultManager object managed
+        state variables in AnalyerState
         """
 
-        model_config_name = run_config_measurement.model_variants_name()
-        model_specific_pa_params, batch_sizes, concurrencies = self._tabulate_measurement_setup(
-            run_config_measurement)
-
-        satisfies = "Yes" if passes else "No"
-
-        # Non GPU specific data
-        inference_fields = self._inference_output_fields
-        inference_row = self._get_common_row_items(
-            inference_fields, batch_sizes, concurrencies, satisfies, model_name,
-            model_config_name, dynamic_batchings, instance_groups,
-            max_batch_sizes, backend_parameters)
-
-        self._populate_inference_rows(run_config_measurement, inference_fields,
-                                      inference_row)
-
-        self._result_tables[self.model_inference_table_key].insert_row_by_index(
-            inference_row)
-
-        # GPU specific data (only put measurement if not cpu only)
-        if not any(cpu_onlys):
-            for gpu_uuid, metrics in run_config_measurement.gpu_data().items():
-                gpu_fields = self._gpu_output_fields
-
-                gpu_row = self._get_common_row_items(
-                    gpu_fields, batch_sizes, concurrencies, satisfies,
-                    model_name, model_config_name, dynamic_batchings,
-                    instance_groups, max_batch_sizes)
-
-                self._add_uuid_to_gpu_row(gpu_row, gpu_uuid, gpu_fields)
-                self._add_metrics_to_gpu_row(gpu_row, metrics, gpu_fields)
-
-                self._result_tables[
-                    self.model_gpu_table_key].insert_row_by_index(row=gpu_row)
+        self._state_manager.set_state_variable('ResultManager.results',
```

Comment:
This class is the only place we should ever use these hardcoded keys, and they should really only do it on one or two places (or use a const)